UiPath.FreeRdpClient/UiPath.SessionTools.Tests/ProcessRunnerExtensionsTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace UiPath.SessionTools.Tests;
+
+using Sut = ProcessRunnerExtensions;
+
+[Trait("Subject", nameof(ProcessRunnerExtensions))]
+[Collection(nameof(ProcessRunnerCollection))]
+public class ProcessRunnerExtensionsTests : ProcessRunnerTestsBase
+{
+    public ProcessRunnerExtensionsTests(ProcessRunnerFixture fixture) : base(fixture) { }
+
+    [Fact(DisplayName = $"{nameof(Sut.SuppressCancellation)} should return a Task that succeeds as soon as the provided Task becomes canceled.")]
+    public async Task SuppressCancellation_ShouldReturnATaskThatSucceeds_WhenTheProvidedTaskBecomesCanceled()
+    {
+        TaskCompletionSource source = new();
+        var task = source.Task.SuppressCancellation();
+
+        task.IsCompletedSuccessfully.ShouldBeFalse();
+
+        source.SetCanceled();
+
+        Should.CompleteIn(
+            task.ShouldNotThrowAsync(),
+            TimeSpan.FromMilliseconds(100));
+    }
+
+    [Fact(DisplayName = $"{nameof(Sut.SuppressCancellation)} should return a Task that succeeds as soon as the provided Task succeeds.")]
+    public async Task SuppressCancellation_ShouldReturnATaskThatSucceeds_WhenTheProvidedTaskSucceeds()
+    {
+        TaskCompletionSource source = new();
+        var task = source.Task.SuppressCancellation();
+
+        task.IsCompletedSuccessfully.ShouldBeFalse();
+
+        source.SetResult();
```

Comment:
doyou mean setException?... maybe also add a test that it also throws, if it does not exists